packages/wagmi/src/hooks/useReadContracts.ts

```diff
@@ -20,7 +21,7 @@ export function useReadContracts<
 ): UseReadContractsReturnType<contracts, allowFailure, selectData> {
   const { watch, ...queryParameters } = parameters
   const queryClient = useQueryClient()
-  const { data: blockNumber } = useBlockNumber({ watch })
+  const { data: blockNumber } = useBlockNumber({ watch, chainId: (parameters.contracts?.[0] as any)?.chainId })
```

Comment:
Actually for use read contracts, I'm not sure if it's a good idea to watch per block as we may have multiple different chain called at the same time?

Reply:
unless we're 100% sure all the calls are made to the same chain